src/NzbDrone.Core/HealthCheck/Checks/RemotePathMappingCheck.cs

```diff
@@ -0,0 +1,98 @@
+using System.Linq;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.EnvironmentInfo;
+using NzbDrone.Common.Messaging;
+using NzbDrone.Core.Download;
+using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.ThingiProvider.Events;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ProviderUpdatedEvent<IDownloadClient>))]
+    [CheckOn(typeof(TrackImportedEvent), CheckOnCondition.FailedOnly)]
```

Comment:
Run this on ProviderAdded and ProviderRemoved event as well? That way these are detected and cleared as soon as client is added or removed.